From 55e59c4b905de2060b32b038032aa6d556f6cf9c Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@frugalware.org>
Date: Sat, 13 Mar 2010 17:21:59 +0100
Subject: [PATCH] Convert ethtool get_stats_count() ops to get_sset_count()

This string query operation was supposed to be replaced by the
generic get_sset_count() starting in 2007.  Convert  r8168's
implementation.
---
 src/r8168_n.c |   13 +++++++++----
 1 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/r8168_n.c b/src/r8168_n.c
index 6bf88a7..05318ce 100644
--- a/src/r8168_n.c
+++ b/src/r8168_n.c
@@ -306,7 +306,7 @@ struct _kc_ethtool_ops {
 	void (*self_test)(struct net_device *, struct ethtool_test *, u64 *);
 	void (*get_strings)(struct net_device *, u32 stringset, u8 *);
 	int  (*phys_id)(struct net_device *, u32);
-	int  (*get_stats_count)(struct net_device *);
+	int  (*get_sset_count)(struct net_device *);
 	void (*get_ethtool_stats)(struct net_device *, struct ethtool_stats *,
 	                          u64 *);
 } *ethtool_ops = NULL;
@@ -1718,9 +1718,14 @@ struct rtl8168_counters {
 };
 
 static int
-rtl8168_get_stats_count(struct net_device *dev)
+rtl8168_get_sset_count(struct net_device *dev, int stringset)
 {
-	return ARRAY_SIZE(rtl8168_gstrings);
+	switch (stringset) {
+	case ETH_SS_STATS:
+		return ARRAY_SIZE(rtl8168_gstrings);
+	default:
+		return -EINVAL;
+	}
 }
 
 static void
@@ -1931,7 +1936,7 @@ static struct ethtool_ops rtl8168_ethtool_ops = {
 	.get_wol		= rtl8168_get_wol,
 	.set_wol		= rtl8168_set_wol,
 	.get_strings		= rtl8168_get_strings,
-	.get_stats_count	= rtl8168_get_stats_count,
+	.get_sset_count		= rtl8168_get_sset_count,
 	.get_ethtool_stats	= rtl8168_get_ethtool_stats,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23)
 #ifdef ETHTOOL_GPERMADDR
-- 
1.7.0.2

